flocker/node/agents/test/test_blockdevice.py

```diff
@@ -1244,6 +1253,27 @@ def __init__(self, dataset_id, node_uuid, mountroot, blockdevice_manager):
         self._testing_mountroot = mountroot
         self._blockdevice_manager = blockdevice_manager
 
+    def _immitate_docker_writing(self, mountdir, filename, content):
+        """
+        Imitate writing content to a file to a mounted directory using docker.
+
+        :param FilePath mountidr: The directory to be mounted as a docker
+            volume.
+        :param unicode filename: The name of the file to create in the docker
+            volume.
+        :param unicode content: The content to write to the file.
+        """
+        container_path = FilePath('/vol')
+        try:
+            check_call([
+                "docker", "run", "--rm", "-v", "%s:%s" % (
+                    mountdir.path, container_path.path),
+                "busybox", "/bin/sh", "-c", "echo -n %s > %s" % (
+                    content, container_path.child(filename).path)
+            ])
+        except CalledProcessError:
```

Comment:
This all drops the output from attempting the write. Would it be possible
to thread it through and include it in the failure message? I imagine
the details of what happened in the command would be important
when trying to debug a failed test.